kiosk/src/Components/ScanQR.tsx

```diff
@@ -53,21 +54,33 @@ const ScanQR: React.FC<IProps> = ({ kiosk }) => {
             <h2>Add game to </h2>
             <h1>Kiosk {kioskId}</h1>
             <div className="scanInstructions">
-                <div className="qrOption">
-                    <p className="scanIntro">Scan game's share QR code with the button below</p>
-                    {
-                        !scannerVisible &&
-                        <button className="scanQrButton" onClick={renderQrScanner} >Scan QR code</button>
-                    }
-                    <div id="qrReader"></div>
-                    {
-                        scannerVisible &&
-                        <p className="scanTip">Tip: Do not use the kiosk's QR code</p>
-                    }
-                </div>
+                {
+                    !linkVisible &&
+                    <div className="qrOption">
+                        <p className="scanIntro">Scan game's share QR code with the button below</p>
+                        {
+                            !scannerVisible &&
+                            <button className="scanQrButton" onClick={renderQrScanner} >Scan QR code</button>
+                        }
+                        <div id="qrReader"></div>
+                        {
+                            scannerVisible &&
+                            <p className="scanTip">Tip: Do not use the kiosk's QR code</p>
+                        }
+                    </div>
+                }
 
                 {
-                    !phoneWidth &&
+                    phoneWidth && !linkVisible &&
+                    <div className="scanIntro">
+                        <p>OR</p>
+                        <button className="scanQrButton back" onClick={() => setLinkVisible(true)} >Submit share link</button>
```

Comment:
would it fit if on mobile, instead of having two 'modes', you only had the `enter share link` input & submit without the extra text / border 
<img width="329" alt="image" src="https://user-images.githubusercontent.com/5615930/208797543-49a26d63-0413-4f05-b92f-7c0ecaee1fe1.png">
and ideally reduce padding / font size on the `add game to kiosk` on mobile so it only takes up one line? it feels like we can show it all at once without too much trouble in that case without having to add an extra 'mode' that might trip them up, and the `can't scan? submit share link here` is a little redundant with the `enter share link` here anyways -- I actually think it'd look good if that portion was slimmed down in favor of showing `OR` on desktop as well -- 

<img width="657" alt="image" src="https://user-images.githubusercontent.com/5615930/208798411-70cad0c2-9c9c-4fb8-a214-39a3c4cd86f9.png">

Reply:
I can slim it down and will add the "or" on the desktop version as I agree, that approach is a bit more clear to the user. 

I will also play with fitting both options on the same page. I have a preference for keeping the two options separate because I couldn't get scrolling to work. When the user goes to scan the QR code, the QR code scanner takes up so much space that the "submit share link" area would not be visible or would be barely visible, and I think that pattern would be off-putting to users. 

I am also not a fan of putting the header on one line as I think it would require the font to be too small and not provide enough visual hierarchy on the page.

Reply:
Fitting both options looks decent. And it makes me realize that I should put a button that allows the user to cancel scanning, anyway. This way, if a user is just playing and seeing what things will do, they can easily get back to the main page to actually enter something if they need to.